internal/topology/check/compatibility.go

```diff
@@ -54,3 +60,95 @@ func ObjectsAreInTheSameNamespace(current, desired client.Object) error {
 	}
 	return nil
 }
+
+// TemplatesAreCompatible checks if two referenced objects are compatible, meaning that
+// they are of the same GroupKind and in the same namespace.
+func TemplatesAreCompatible(current, desired clusterv1.LocalObjectTemplate) error {
```

Comment:
Q: this seems a duplicate of `ReferencedObjectsAreCompatible`. Have you considered to use this func (or to rename it)?

Reply:
Same for `TemplatesAreInTheSameNamespace` vs `ObjectsAreInTheSameNamespace`

Reply:
The issue is the signature here - ReferencedObjects* takes client.Objects while these take LocalObjectTemplates(or possibly ObjectReferences). The alternative - I think at least - is to have some sort of conversion function for the references. I'm happy to do it that way either, but I don't see a completely clean solution.

Reply:
The problem is that they are operating on different objects. 

In general we would need a combination of runtime.Object + `GetNamespace() string`.

We would have to create the interface and implement it with `LocalObjectTemplate`.

I'm not sure if it's worth doing it to avoid the duplication here.

Reply:
We could do a ObjectRefToObject method and just fill in the fields, but it's not ideal either.

Reply:
I have the feeling we'll end up with more code with any of those options.